server/onlineUsers.js

```diff
@@ -1,2 +1,42 @@
-const onlineUsers = [];
+class OnlineUsers {
+  // Stores a dictionary of structure:
+  // {
+  //   userId :
+  //   {
+  //     socketId: int
+  //   }
+  // }
+  constructor() {
+    this.users = {};
+  }
+
+  // checks if userId exists
+  includes(userId) {
+    return userId in this.users;
+  }
+
+  // returns true if added, false if not added
+  addUser(userId, socketId) {
+    if (this.includes(userId)) return false;
+    this.users[userId] = {
+      socketId: socketId,
+    };
+    return true;
+  }
+
+  // returns the user structure
+  // null if it does not exist
+  getUser(userId) {
+    if (this.includes(userId)) {
```

Comment:
Should this be  (userId in this.users) or similar?  I haven't this this.includes on an object before (typically an array method).